tests/test_asset_list.py

```diff
@@ -0,0 +1,232 @@
+        with self.assertRaises(fv3config.ConfigError):
+            ensure_is_list(sample_float)
+
+    def test_generate_asset_default_options(self):
+        test_asset = generate_asset(SAMPLE_SOURCE_LOCATION, SAMPLE_SOURCE_NAME)
+        self.assertEqual(test_asset, SAMPLE_ASSET_NO_KWARGS)
+
+    def test_generate_asset_custom_options(self):
+        test_asset = generate_asset(SAMPLE_SOURCE_LOCATION, SAMPLE_SOURCE_NAME,
+                                    target_location=SAMPLE_TARGET_LOCATION,
+                                    target_name=SAMPLE_TARGET_NAME,
+                                    copy_method=SAMPLE_COPY_METHOD)
+        self.assertEqual(test_asset, SAMPLE_ASSET_WITH_KWARGS)
+
+    def test_asset_list_from_local_dir_empty(self):
+        workdir = self.make_work_directory('workdir_empty')
+        self.make_empty_files(workdir, FILELIST_EMPTY)
+        asset_list = asset_list_from_local_dir(workdir)
+        self.assertEqual(asset_list, ASSET_LIST_EMPTY)
+
+    def test_asset_list_from_local_dir_one_file_in_root(self):
+        workdir = self.make_work_directory('workdir_with_files')
+        self.make_empty_files(os.path.join(workdir, TEST_SOURCE_DIR),
+                              FILELIST_ONE_FILE_IN_ROOT)
+        asset_list = asset_list_from_local_dir(os.path.join(workdir, TEST_SOURCE_DIR))
+        # necessary to have full path for source location in test assets
+        for asset in ASSET_LIST_ONE_FILE_IN_ROOT:
+            asset['source_location'] = os.path.join(workdir, asset['source_location'])
+        self.assertEqual(asset_list, ASSET_LIST_ONE_FILE_IN_ROOT)
+
+    def test_asset_list_from_local_dir_with_subdir(self):
+        workdir = self.make_work_directory('workdir_with_files')
+        self.make_empty_files(os.path.join(workdir, TEST_SOURCE_DIR), FILELIST_WITH_SUBDIR)
+        asset_list = asset_list_from_local_dir(os.path.join(workdir, TEST_SOURCE_DIR))
+        # necessary to have full path for source location in test assets
+        for asset in ASSET_LIST_WITH_SUBDIR:
+            asset['source_location'] = os.path.join(workdir, asset['source_location'])
+        self.assertEqual(asset_list, ASSET_LIST_WITH_SUBDIR)
+
+    @staticmethod
+    def make_empty_files(directory, filelist):
+        for path in filelist:
+            full_path = os.path.join(directory, path)
+            head, tail = os.path.split(full_path)
+            if not os.path.exists(head):
+                os.makedirs(head)
+            open(full_path, 'a').close()
+
+    def test_check_asset_valid_bad_asset(self):
+        bad_asset_dict = {'irrelevant_key': ''}
+        with self.assertRaises(fv3config.ConfigError):
+            check_asset_valid(bad_asset_dict)
+
+    def test_check_asset_valid_proper_asset(self):
+        proper_asset_dict = {
+            'source_location': '',
+            'source_name': '',
+            'target_location': '',
+            'target_name': '',
+            'copy_method': '',
```

Comment:
Is this really a valid asset? copy_method is not one of the valid options, and presumably the names need to be non-empty.

Reply:
At some point in the future we may want to use something cool [like `JSON-Schema`](https://stackoverflow.com/questions/3262569/validating-a-yaml-document-in-python) to validate our dictionaries.

Reply:
Changed function name to `check_asset_has_required_keys` to better reflect its purpose.

Reply:
I suppose we'll refactor this test later if we implement more stringent checks.

Reply:
I agree, this could be done as part of implementing #6 and #22